### ISSUE TITLE
setAttribute error

### ISSUE THREAD

**Issue by kristoferjoseph** [graforlock/diffing-implementation#2]
https://github.com/graforlock/diffing-implementation/blob/master/lib/patch.js#L121
`oldNode.attributes[i]` could be undefined so accessing value can error.

**Comment by graforlock** [comment]
Yeah, you're right, the implementation is sorta rough draft. Feel free to make amends and PR.

**Comment by kristoferjoseph** [comment]
@graforlock i totally understand, was just trying to help out by tracking down any issues i was seeing. Will make some PRs when i find some time.

This is really great work btw. Thanks for taking to the time to rough it out. 🙌
